src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests.Enumeration
+{
+    public class TrimmedPaths : FileSystemTest
+    {
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TrimmedPathsAreFound_Windows()
+        {
+            // Trailing spaces and periods are eaten when normalizing in Windows, making them impossible
+            // to access without using the \\?\ device syntax. We should, however, be able to find them
+            // and retain the filename in the info classes and string results.
+
+            var directory = Directory.CreateDirectory(GetTestFilePath());
+            File.Create(@"\\?\" + Path.Combine(directory.FullName, "Trailing space ")).Dispose();
+            File.Create(@"\\?\" + Path.Combine(directory.FullName, "Trailing period.")).Dispose();
+
+            var files = directory.GetFiles();
+            Assert.Equal(2, files.Count());
+            FSAssert.EqualWhenOrdered(new string[] { "Trailing space ", "Trailing period." }, files.Select(f => f.Name));
+
+            var paths = Directory.GetFiles(directory.FullName);
+            Assert.Equal(2, paths.Count());
+            FSAssert.EqualWhenOrdered(new string[] { "Trailing space ", "Trailing period." }, paths.Select(p => Path.GetFileName(p)));
+        }
```

Comment:
>} [](start = 8, length = 1)

Q.: just to confirm: is the practice to leave such files on the disk? I know that they don't have data but I want to confirm the practice here.

Reply:
Hum reading the test below it seems that the first is redundant and it actually deletes the files...

---
In reply to: [166747719](https://github.com/dotnet/corefx/pull/26918#discussion_r166747719) [](ancestors = 166747719)

Reply:
The base class will clean up. It does overlap with the other test, but this discrete one will help navigate to failures a bit quicker.